File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let frequency = {}

        for(let num of nums){
            frequency[num] = (frequency[num]||0)+1
        }

        const sorted = Object.keys(frequency).sort((a,b)=>frequency[b]-frequency[a])
        return sorted.slice(0,k).map(Number)
    }
    
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        s = 0
        e = n-1

        while s<=e:
            mid = (s+e)//2
            if target == nums[mid]:
                return mid
            elif target > nums[mid]:
                s+=1
            else:
                e-=1
        return -1